packages/collector/src/announceCycle/unannounced.js

```diff
@@ -86,19 +92,26 @@ function tryToAnnounce(ctx, retryDelay = initialRetryDelay) {
     try {
       agentResponse = JSON.parse(rawResponse);
     } catch (e) {
-      logger.warn(
-        'Failed to JSON.parse agent response. Response was %s. Will retry in %s ms',
-        rawResponse,
+      logger.error(
+        "Failed to parse the JSON payload from the Instana host agent's response. Establishing the " +
+          'connection to the Instana host agent will be retried in %s ms. The response payload was %s.',
         retryDelay,
+        rawResponse,
         e
       );
       setTimeout(tryToAnnounce, retryDelay, ctx, nextRetryDelay).unref();
       return;
     }
 
-    const pid = agentResponse.pid;
-    logger.info('Overwriting pid for reporting purposes to: %s', pid);
-    pidStore.pid = pid;
+    if (pidStore.pid !== agentResponse.pid) {
+      logger.info(
+        'Reporting data to the Instana host agent with the PID from the root namespace (%s) instead of the ' +
+          'in-container PID (%s).',
+        agentResponse.pid,
+        pidStore.pid
+      );
+      pidStore.pid = agentResponse.pid;
```

Comment:
This could be considered a change in behavior, but in practical terms it should not. We now only overwrite the PID when the PID returned by the agent is actually different. (And also log the fact only in that case.)